OpenRA.Mods.Common/Traits/Buildings/GivesBuildableArea.cs

```diff
@@ -0,0 +1,35 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Tag trait for actors with `AreaTypes`.")]
+	public class GivesBuildableAreaInfo : ConditionalTraitInfo
+	{
+		[Desc("Given area types to be checked by `Building` traits of actors while being placed.")]
+		public readonly HashSet<string> Types = new HashSet<string>() { "building" };
```

Comment:
Also can you change this to `AreaTypes` to match the one on `Building`/`RequiresBuildableArea` and because upgrade rules can be a pain to write sometimes.